File: bench/ack.rb
require 'benchmark'
def Ack(m, n)
    if (m == 0) then
        return n + 1
    end
    if (n == 0) then
        return Ack(m - 1, 1)
    end
    return Ack(m - 1, Ack(m, (n - 1)))
end

time = Benchmark.measure{
puts Ack(3,9)
}

puts time.real*1000

File: bench/ack.js
function Ack(M, N) {
    if (M === 0)
        return N + 1
    if (N === 0)
        return Ack(M - 1, 1)
    return Ack(M - 1, Ack(M, (N - 1)))
}

let time = performance.now()
console.log(Ack(3, 9))
console.log(performance.now() - time)